packages/sdk/src/actions/simulateUpdateDepositTx.ts

```diff
@@ -0,0 +1,184 @@
+ *   },
+ *   update: {
+ *     recipient: "0xNEW_RECIPIENT_ADDRESS",
+ *   },
+ *  });
+ * const txHash = await walletClient.writeContract({
+ *   account,
+ *   ...txRequest,
+ * });
+ * ```
+ */
+export async function simulateUpdateDepositTx(
+  params: SimulateUpdateDepositTxParams,
+) {
+  const {
+    walletClient,
+    deposit: _deposit,
+    update,
+    originChainClient,
+    destinationChainClient,
+    logger,
+    apiUrl,
+  } = params;
+
+  const {
+    originChainId,
+    originSpokePoolAddress,
+    destinationChainId,
+    destinationSpokePoolAddress,
+    depositId,
+  } = _deposit;
+
+  const account = walletClient.account;
+
+  if (!account) {
+    throw new Error("Wallet account has to be set");
+  }
+
+  const connectedChainId = await walletClient.getChainId();
+
+  if (connectedChainId !== originChainId) {
+    throw new Error(
+      `Connected chainId ${connectedChainId} does not match originChainId ${originChainId}`,
+    );
+  }
+
+  const prevDeposit = await getDeposit({
+    originChainClient,
+    destinationChainClient,
+    findBy: {
+      depositId,
+      originChainId,
+      destinationChainId,
+      destinationSpokePoolAddress,
+    },
+  });
+
+  if (prevDeposit.depositor !== account.address) {
+    throw new Error(
+      `Connected account ${account.address} does not match depositor ${prevDeposit.depositor}`,
+    );
+  }
+
+  const updatedRecipient = update.recipient ?? prevDeposit.recipient;
+
+  const newQuote = await getQuote({
+    route: {
+      originChainId,
+      destinationChainId,
+      inputToken: prevDeposit.inputToken,
+      outputToken: prevDeposit.outputToken,
+    },
+    inputAmount: prevDeposit.inputAmount,
+    recipient: updatedRecipient,
+    crossChainMessage: update.crossChainMessage ?? prevDeposit.message,
+    logger,
+    apiUrl,
+  });
+
+  if (
+    update.outputAmount &&
+    update.outputAmount < newQuote.deposit.outputAmount
+  ) {
+    throw new Error(
+      `Provided 'update.outputAmount' ${update.outputAmount} is less than the required ${newQuote.deposit.outputAmount}`,
+    );
+  }
+
+  const updatedOutputAmount =
+    update.outputAmount ?? newQuote.deposit.outputAmount;
+  const updatedMessage = newQuote.deposit.message;
+  const signature = await walletClient.signTypedData({
+    account: account.address,
+    domain: {
+      name: "ACROSS-V2",
+      version: "1.0.0",
+      chainId: originChainId,
+    },
+    types: {
+      UpdateDepositDetails: [
+        { name: "depositId", type: "uint32" },
+        { name: "originChainId", type: "uint256" },
+        { name: "updatedOutputAmount", type: "uint256" },
+        { name: "updatedRecipient", type: "address" },
+        { name: "updatedMessage", type: "bytes" },
+      ],
+    },
+    primaryType: "UpdateDepositDetails",
+    message: {
+      depositId,
+      originChainId: BigInt(originChainId),
+      updatedOutputAmount,
+      updatedRecipient,
+      updatedMessage,
+    },
+  });
```

Comment:
@dohaki Should we factor this out to a separate function? I think it could be generically useful; we've previously had cases where people wanted to produce this signature and we didn't have the tooling available.

Reply:
Yes good point! Will do

Reply:
Extracted here https://github.com/across-protocol/toolkit/pull/47/commits/f8a54a4fb29f5e85a78246a35997021bd84e589f